kusd/validator/states.go

```diff
@@ -91,6 +52,60 @@ func (val *Validator) notLoggedInState() stateFn {
 	return val.newElectionState
 }
 
+func (val *Validator) nodeNeedsToDeposit() bool {
+	// @NOTE (rgeraldes) - sync was already done at this point and by default the investors will be
+	// part of the initial set of validators - no need to make a deposit if the block number is 0
+	// since these validators will be marked as voters from the start
+	isGenesis, err := val.network.IsGenesisVoter(&bind.CallOpts{}, val.account.Address)
+	if err != nil {
+		log.Crit("Failed to verify the voter information", "err", err)
+	}
+	return !isGenesis || val.isGenesisRejoining(isGenesis)
+}
+
+func (val *Validator) isGenesisRejoining(isGenesis bool) bool {
+	return isGenesis && val.isNotFirstBlock()
+}
+
+func (val *Validator) isFirstBlock() bool {
+	return val.chain.CurrentBlock().NumberU64() == 0
+}
+
+func (val *Validator) isNotFirstBlock() bool {
```

Comment:
I think using a `!` to negate the `isFirstBlock` looks better. If in the future some is refactoring a code where at first it was using `isNotFirstBlock` and the new logic requires checking for first block, I doubt that person will think there's another function with the negative and it'd use `!val.isNotFirstBlock()`. It'd happen to me :P

Reply:
this is from clean code, readability first. but I'm not too fussed about it. They say use the one that is most common (usage)

Reply:
I see. Then, all `is...` functions should have the `isNot...` as well? It's not something very common, I've only seen this kind of things in testing libraries for assertions, but not for general code.

Reply:
Not necessarily all functions should have the Not,
The main point is readability, in some places the ! is easy to misses at first look
I'm being a bit nitpicking :) I will remove one